lib/core/test/unit/Cardano/Pool/RankingSpec.hs

```diff
@@ -0,0 +1,284 @@
+    -- TODO: set pool{R.stake = stake}, but that field doesn't even exist!
+
+data EffectOnDesirability
+    = PositiveEffect
+    | NegativeEffect
+    deriving (Show, Eq)
+
+-- | All else equal, what effect on the desirability does changing a single
+-- field have?
+allElseEqualProperty
+    :: forall (field :: Symbol) a.
+        ( Arbitrary a
+        , Show a
+        , Ord a
+        , HasField' field Pool a
+        , KnownSymbol field
+        )
+    => EffectOnDesirability
+        -- ^ Expected effect on the desirability
+    -> Pool
+       -- ^ The base pool which is later modified
+    -> EpochConstants
+       -- ^ Constants used
+    -> Property
+allElseEqualProperty expectedEffect pool constants =
+    forAll arbitrary $ \(value1 :: a) (value2 :: a) -> do
+        let (higherVal, lowerVal) = sortTupleOn id (value1, value2)
+
+        let (moreDesPool, lessDesPool) = sortTupleOn snd
+                ( poolDesPairWith value1
+                , poolDesPairWith value2
+                )
+
+        let poolRankDesc = unlines
+                [ "Pools:"
+                , showPool moreDesPool
+                , showPool lessDesPool
+                ]
+
+        case des higherVal `compare` des lowerVal of
+            EQ -> property True
+            -- A reason we end up in this case could be that both pools are
+            -- un-profitable. It would be interesting to re-run tests with
+            -- a higher total reward pot. But as long we hit the other cases we
+            -- should be fine, really.
+            GT -> expectedEffect === PositiveEffect
+            LT -> expectedEffect === NegativeEffect
+          & counterexample poolRankDesc
+          & classify (des higherVal == des lowerVal) "same desirability"
+  where
+    poolWith :: a -> Pool
+    poolWith val = pool & (field' @field) .~ val
+
+    poolDesPairWith :: a -> (Pool, Double)
+    poolDesPairWith val = (poolWith val, desirability constants (poolWith val))
+
+    sortTupleOn :: (Ord c) => (b -> c) -> (b, b) -> (b, b)
+    sortTupleOn f (a, b)
+        | (f a) > (f b) = (a, b)
+        | otherwise     = (b, a)
+
+    showPool :: (Pool, Double) -> String
+    showPool (x, d) = unwords
+        [ show (symbolVal $ Proxy @field)
+        , "="
+        , show (x ^. field' @field)
+        , ", desirability ="
+        , show d
+        ]
+
+    des val = desirability constants (poolWith val)
+
+instance Arbitrary Pool where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
+
+-- TODO: We should ideally not export the NonNegative and Positive constructors,
+-- in which case we wouldn't be able to use DerivingVia.
+deriving via Word64 instance (Arbitrary Lovelace)
+deriving via (NonNegative Double) instance (Arbitrary (R.NonNegative Double))
+deriving via (Positive Int) instance (Arbitrary (R.Positive Int))
+
+instance Arbitrary Ratio where
+    arbitrary = unsafeMkRatio <$> choose (0, 1)
+    shrink _ = [] -- TODO: Implement a shrinker that works.
```

Comment:
It was the `Ratio` shrinker that was faulty. I removed it, and re-added `genericShrink` for `Pool`.